src/App.tsx

```diff
@@ -7,33 +8,60 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    document.addEventListener('mouseup', this.handleClick);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    this.timerId = window.setInterval(() => {
+      this.setState({
+        clockName: getRandomName(),
+      });
+    }, 3000);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount() {
+    document.removeEventListener('mouseup', this.handleClick);
+    window.clearInterval(this.timerId);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  handleClick = (event: MouseEvent) => {
+    if (event.button === 0) {
+      this.setState({
+        hasClock: true,
+      });
+    }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    if (event.button === 2) {
```

Comment:
Why do you need this check? I think it's redundant.
And you should separate this logic to different function.